### PR TITLE
Containerized molecule

### PR DESCRIPTION
**Pull request by pthomison**
- Makefile
- Dockerfile
- Tweaks

cp-server is still losing its find, but not sure why yet

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+current_dir = $(shell pwd)
+# uid = $(shell id -u)
+# gid = $(shell id -g)
```
Comment (domenicbove) [inline Makefile]:
```suggestion
```

File: roles/ssl/tasks/main.yml [hunk 2]
```diff
@@ -86,7 +92,8 @@
       -in {{keystore_path}} \
       -passin pass:{{keystore_storepass}} \
       -nodes -nocerts \
-      -out {{key_path}}
+      -out {{key_path}} && \
+    chown -R {{uid}}:{{gid}} {{key_path}}
   args:
     # Assuming the key is correct if exists
     creates: "{{key_path}}"
```
Comment (domenicbove) [inline roles/ssl/tasks/main.yml]: WDYT about removing all the chown commands and putting this as the final task:
```
- name: Create Directory for SSL Files
  file:
    path: "{{component_output_dir}}"
    recursive: true
    owner: "{{ uid }}"
    group: "{{ gid }}"
```

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+define molecule_docker_exec 
+docker run \
+-it --rm \
+-v "/var/run/docker.sock:/var/run/docker.sock" \
+-v "$(current_dir):$(current_dir)" \
+-w "$(current_dir)" \
+--entrypoint="" \
+-e "HOME=/tmp/" \
```
Comment (domenicbove) [inline Makefile]:
```suggestion
```
are these necessary

File: roles/ssl/tasks/main.yml (hunk 2 above)
Comment (pthomison) [reply roles/ssl/tasks/main.yml]: Works for me! component_output_dir will be valid for both local and remote deployments right?

File: Makefile (hunk 3 above)
Comment (pthomison) [reply Makefile]: Nah they can be dropped. HOME is cruft left over from the UID fun && the entrypoint for the image is already set to sh

**Review by domenicbove — APPROVED** [review]
the docker run command is awesome! I think im going to update the readme to just have that instead of molecule

**Pull request merged**